hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -144,22 +146,41 @@ case class HoodieFileIndex(spark: SparkSession,
     // Prune the partition path by the partition filters
     // NOTE: Non-partitioned tables are assumed to consist from a single partition
     //       encompassing the whole table
-    val prunedPartitions = listMatchingPartitionPaths(partitionFilters)
+    val prunedPartitions = if (shouldBroadcast) {
+      listMatchingPartitionPaths(convertFilterForTimestampKeyGenerator(metaClient, partitionFilters))
+    } else {
+      listMatchingPartitionPaths(partitionFilters)
+    }
     val listedPartitions = getInputFileSlices(prunedPartitions: _*).asScala.toSeq.map {
       case (partition, fileSlices) =>
-        val baseFileStatuses: Seq[FileStatus] = getBaseFileStatus(fileSlices
+        var baseFileStatuses: Seq[FileStatus] = getBaseFileStatus(fileSlices
           .asScala
-          .map(fs => fs.getBaseFile.orElse(null))
-          .filter(_ != null))
-
+            .map(fs => fs.getBaseFile.orElse(null))
+            .filter(_ != null))
+        if (shouldBroadcast) {
+          baseFileStatuses = baseFileStatuses ++ fileSlices.asScala
```

Comment:
Rename this variable since this contains `FileStatus` instance from file slices with only log files?